.github/workflows/ci.yml

```
@@ -24,35 +23,42 @@ jobs:
             ghc: 8.8.4
           - os: macOS-latest
             ghc: 8.6.5
-          - os: windows-latest
-            ghc: 8.8.4
           - os: windows-latest
             ghc: 8.10.2
+          - os: windows-latest
+            ghc: 8.6.5
 
     steps:
     - uses: actions/checkout@v2
       if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/main'
 
-    - uses: actions/setup-haskell@v1.1.3
+    - uses: actions/setup-haskell@v1.1.4
       id: setup-haskell-cabal
       name: Setup Haskell
       with:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ matrix.cabal }}
 
+    - name: Configure
+      run: |
+        cabal configure --enable-tests --enable-benchmarks --test-show-details=direct
+
     - name: Freeze
       run: |
         cabal freeze
 
-    - uses: actions/cache@v2.1.1
+    - uses: actions/cache@v2.1.3
       name: Cache ~/.cabal/store
       with:
         path: ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
         key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
 
+    - name: Install dependencies
+      run: |
+        cabal build all --only-dependencies
```

I wonder if it makes sense to change all steps to single-line commands? 🤔 It will make the config a bit smaller (like, 5 lines smaller). But probably will be confusing for people, who want to add new commands...

```suggestion
      run: cabal build all --only-dependencies
```